codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -92,8 +96,12 @@ public void run() {
                         writer.write("if err := fn(stack); err != nil { return nil, err }");
                     });
 
-                    constructHandler();
+                    if (protocolGenerator != null) {
+                        // add middleware to operation stack
```

Comment:
Shouldn't this just be a generic GoIntegration extension point? This would allow other handlers to be registered in a generic way from plugins.

Reply:
Looks like this PR mistakenly included changes that it shouldn't of. this is a part of https://github.com/awslabs/smithy-go/pull/66 will update the pr to not include those extra changes.